.github/workflows/test.yml

```
@@ -196,6 +196,83 @@ jobs:
           files: ./packages/client/src/__tests__/coverage/clover.xml
           flags: client,${{ matrix.os }},${{ matrix.queryEngine }}
           name: client-${{ matrix.os }}-${{ matrix.queryEngine }}
+
+  #
+  # CLIENT (functional tests with mini-proxy)
+  #
+  client-dataproxy:
+    timeout-minutes: 35
+    runs-on: ${{ matrix.os }}
+
+    needs: detect_jobs_to_run
+    if: ${{ contains(needs.detect_jobs_to_run.outputs.jobs, '-all-') || contains(needs.detect_jobs_to_run.outputs.jobs, '-client-') }}
+
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [buildjet-4vcpu-ubuntu-2004]
+        node: [14, 16, 18]
```

```suggestion
        node: [14, 16, 18]
        type: ['--data-proxy', '--data-proxy --edge-client']
```

Could we leverage a matrix here? It looks like it's only the command-like arguments that are changing... see other suggestions.

---

This is a nice idea!

---

Thank you!

---

Ah, actually I think I know why I didn't do this (I didn't think of leveraging the matrix but I did think about making these separate jobs manually): that might create a bit too many CI jobs and exhaust the number of runners. Let's leave it for now, and split it later if we see that a single job takes too long.